mlflow/recipes/utils/metrics.py

```diff
@@ -84,17 +84,21 @@ def _get_error_fn(tmpl: str, use_probability: bool = False, positive_class: Opti
         return lambda predictions, targets: predictions - targets
     if tmpl == "classification/v1":
         if use_probability:
-            import numpy as np
 
-            def logloss(true_label, predicted_proba, eps=1e-15):
-                p = np.clip(predicted_proba, eps, 1 - eps)
+            def error_rate(true_label, predicted_positive_class_proba, eps=1e-15):
                 if true_label == positive_class:
-                    return -np.log(p)
+                    # if true_label == positive_class then the probability is
+                    # predicted_positive_class_proba but the error rate is
+                    # 1 - predicted_positive_class_proba
+                    return 1 - predicted_positive_class_proba
                 else:
-                    return -np.log(1 - p)
+                    # if true_label != positive_class then the probability is
+                    # 1 - predicted_positive_class_proba but the error rate is
+                    # predicted_positive_class_proba
+                    return predicted_positive_class_proba
 
             return lambda predictions, targets: [
-                logloss(x, y) for (x, y) in zip(targets, predictions[:, 0])
+                error_rate(x, y) for (x, y) in zip(targets, predictions)
```

Comment:
@sunishsheth2009 why do we need to change the calculation logic away from logloss? i'm not following...maybe an example would help

Reply:
Basically, we care about absolute error based on probabilities. 
So in the above case, imagine this usecase:
positive_class = 1

for 1st given row:
probability_1 : 0.8
label: 1
predicted: 1
So ideally the error rate where label == positive_class: error rate is 1-0.8 = 0.2

for 2nd given row:
probability_1 : 0.2
label: 1
predicted: 0
So ideally the error rate where label == positive_class: error rate is 1-0.2 = 0.8

for 3rd given row:
probability_1 : 0.8
label: 0
predicted: 1
So ideally the error rate where label != positive_class: error rate is 0.8

for 4rd given row:
probability_1 : 0.2
label: 0
predicted: 0
So ideally the error rate where label != positive_class: error rate is 0.2

Is this helpful?

Reply:
How does it work in multi-class scenario, can you give a similar example with 3-classes `[0, 1, 2]`, with `positive_class = 1`? Specificallly,
```
for 4rd given row:
probability_1 : 0.2
label: 0
predicted: 0
```
What's the error rate here?

Reply:
We don't support in multi-class since currently multi-class doesn't have an option of `positive_class`. If we add `positive_class` with >2 cardinality, we throw an error.

Reply:
Can we add comment here to indicate that only binary classification is supported for worst examples for future readers?

Reply:
Added a comment :)